src/services/dumpers/agent-nodejs.ts

```diff
@@ -0,0 +1,209 @@
+  private env: { FOREST_SERVER_URL: string; FOREST_URL_IS_DEFAULT: boolean };
+
+  private readonly DEFAULT_PORT = 3310;
+
+  private readonly isLinuxOs: boolean;
+
+  private readonly isDatabaseLocal: (dbConfig: DbConfig) => boolean;
+
+  private readonly buildDatabaseUrl: (dbConfig: DbConfig) => string;
+
+  private readonly lodash: typeof Lodash;
+
+  private readonly toValidPackageName: (string: string) => string;
+
+  constructor(context) {
+    const {
+      assertPresent,
+      env,
+      isLinuxOs,
+      buildDatabaseUrl,
+      isDatabaseLocal,
+      lodash,
+      toValidPackageName,
+    } = context;
+
+    assertPresent({
+      env,
+      isLinuxOs,
+      buildDatabaseUrl,
+      isDatabaseLocal,
+      lodash,
+      toValidPackageName,
+    });
+
+    super(context);
+
+    this.env = env;
+    this.isLinuxOs = isLinuxOs;
+    this.buildDatabaseUrl = buildDatabaseUrl;
+    this.isDatabaseLocal = isDatabaseLocal;
+    this.lodash = lodash;
+    this.toValidPackageName = toValidPackageName;
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  protected get templateFolder() {
+    return 'agent-nodejs';
+  }
+
+  writePackageJson(dbDialect: string, appName: string) {
+    const dependencies: { [name: string]: string } = {
+      dotenv: '^16.0.1',
+      '@forestadmin/agent': '^1.0.0',
+    };
+
+    if (dbDialect === 'mongodb') {
+      dependencies['@forestadmin/datasource-mongoose'] = '^1.0.0';
+    } else {
+      dependencies['@forestadmin/datasource-sql'] = '^1.0.0';
+    }
+
+    if (dbDialect) {
+      if (dbDialect.includes('postgres')) {
+        dependencies.pg = '~8.2.2';
+        dependencies['pg-hstore'] = '~2.3.4';
+      } else if (dbDialect === 'mysql') {
+        dependencies.mysql2 = '~2.2.5';
+      } else if (dbDialect === 'mariadb') {
+        dependencies.mariadb = '^2.3.3';
+      } else if (dbDialect === 'mssql') {
+        dependencies.tedious = '^6.4.0';
+      }
+    }
+
+    const pkg = {
+      name: this.toValidPackageName(appName),
+      version: '0.0.1',
+      private: true,
+      main: 'index.js',
+      scripts: {
+        'start:watch': 'nodemon',
+        start: 'node ./index.js',
+      },
+      nodemonConfig: {
+        ignore: ['./forestadmin-schema.json'],
+      },
+      dependencies,
+      devDependencies: {
+        nodemon: '^2.0.12',
+      },
+    };
+
+    this.writeFile('package.json', `${JSON.stringify(pkg, null, 2)}\n`);
+  }
+
+  writeIndex(dbDialect: string) {
+    const isMongoose = dbDialect === 'mongodb';
+
+    const context = {
+      isMongoose,
+      isMySQL: dbDialect === 'mysql',
+      isMSSQL: dbDialect === 'mssql',
+      isMariaDB: dbDialect === 'mariadb',
+      forestServerUrl: this.env.FOREST_URL_IS_DEFAULT ? false : this.env.FOREST_SERVER_URL,
+      datasourceImport: isMongoose
+        ? `const { createMongooseDataSource } = require('@forestadmin/datasource-mongoose');\nconst connection = require('./mongoose-models');`
```

Comment:
Is mongoose code relevant here? `mongoose-models` will not exist in this PR, right ?

Reply:
Yes, it won't exist. However that's what is defined in the epic, the goal is to dump the main files in this PR - meaning the index.js. This is not a PR with functionnal impact, as long as the dumpers are released before the commands it won't be an issue (and again it's the release order from the PRs). 
I can move it to the next PR if you want but it'll take some time.

Reply:
This PR is not meant to be released alone, it will be merged once and only once PR for mongoose models will be merged into this one. 

We understand the fact that mongoose datasource has nothing to do here, however, this will be overwritten after the merge.